src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_RANGE = 20;
+    private final Random random = new Random();
+    private final Circle circle = new Circle();
+    private final Square square = new Square();
+    private final RightTriangle rightTriangle = new RightTriangle();
+    private final Rectangle rectangle = new Rectangle();
+    private final IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid();
+    private final ColorSupplier colorSupplier = new ColorSupplier(random);
+    private final Figure[] figures = {circle, square, rightTriangle, rectangle, isoscelesTrapezoid};
+
+    public Figure getRandomFigure() {
+        int randomNameFigure = random.nextInt(figures.length);
+        switch (randomNameFigure) {
+            case 0:
+                circle.setColor(colorSupplier.getRandomColor());
+                circle.setRadius(random.nextInt(MAX_RANGE) + 1);
+                return circle;
```

Comment:
```suggestion
                Circle circle = new Circle();
                circle.setColor(colorSupplier.getRandomColor());
                circle.setRadius(random.nextInt(MAX_RANGE) + 1);
                return circle;
```

Applicable for all cases. So move from class-level declaration here.